optuna/storages/_rdb/alembic/versions/v3.0.0.d.py

```diff
@@ -0,0 +1,167 @@
+
+
+BaseModel = declarative_base()
+RDB_MAX_FLOAT = np.finfo(np.float32).max
+RDB_MIN_FLOAT = np.finfo(np.float32).min
+
+
+FLOAT_PRECISION = 53
+
+
+class TrialValueModel(BaseModel):
+    class TrialValueType(enum.Enum):
+        FINITE = 1
+        INF_POS = 2
+        INF_NEG = 3
+
+    __tablename__ = "trial_values"
+    trial_value_id = sa.Column(sa.Integer, primary_key=True)
+    value = sa.Column(sa.Float(precision=FLOAT_PRECISION), nullable=True)
+    value_type = sa.Column(sa.Enum(TrialValueType), nullable=False)
+
+    @classmethod
+    def value_to_stored_repr(
+        cls,
+        value: float,
+    ) -> Tuple[Optional[float], TrialValueType]:
+        if value == float("inf"):
+            return (None, cls.TrialValueType.INF_POS)
+        elif value == float("-inf"):
+            return (None, cls.TrialValueType.INF_NEG)
+        else:
+            return (value, cls.TrialValueType.FINITE)
+
+    @classmethod
+    def stored_repr_to_value(cls, value: Optional[float], float_type: TrialValueType) -> float:
+        if float_type == cls.TrialValueType.INF_POS:
+            assert value is None
+            return float("inf")
+        elif float_type == cls.TrialValueType.INF_NEG:
+            assert value is None
+            return float("-inf")
+        else:
+            assert float_type == cls.TrialValueType.FINITE
+            assert value is not None
+            return value
+
+
+def upgrade():
+    bind = op.get_bind()
+
+    sa.Enum(TrialValueModel.TrialValueType).create(bind, checkfirst=True)
+
+    # MySQL and PostgreSQL supports DEFAULT clause like 'ALTER TABLE <tbl_name>
+    # ADD COLUMN <col_name> ... DEFAULT "FINITE"', but seemingly Alembic
+    # does not support such a SQL statement. So first add a column with schema-level
+    # default value setting, then remove it by `batch_op.alter_column()`.
+    with op.batch_alter_table("trial_values") as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "value_type",
+                sa.Enum("FINITE", "INF_POS", "INF_NEG", name="trialvaluetype"),
+                nullable=False,
+                server_default="FINITE",
+            ),
+        )
+    with op.batch_alter_table("trial_values") as batch_op:
+        batch_op.alter_column("value_type", server_default=None)
+        batch_op.alter_column(
+            "value",
+            existing_type=sa.Float(precision=FLOAT_PRECISION),
+            existing_nullable=False,
+            nullable=True,
+        )
+
+    session = orm.Session(bind=bind)
+    try:
+        records = session.query(TrialValueModel).all()
+        mapping = []
+        for r in records:
+            value: float
+            if np.isclose(r.value, RDB_MAX_FLOAT) or np.isposinf(r.value):
+                value = float("inf")
+            elif np.isclose(r.value, RDB_MIN_FLOAT) or np.isneginf(r.value):
+                value = float("-inf")
+            elif np.isnan(r.value):
+                value = float("nan")
```

Comment:
[Notes] Trial values do not accept `NaN`, but it may happen to be stored in the storage (e.g., during trial-and-error of optuna codebase). This code simply pass such NaN value as `FINITE` value type. I guess it is acceptable, but if we strictly reject `NaN` value, we should raise errors here.

Reply:
I talked with @contramundum53 and @HideakiImamura , and they proposed to remove this `nan` handling since `nan` value is not expected here.

Reply:
Thank you for pointing that out! I forgot to remove those lines.
I fixed it now.

Reply:
Thanks!